app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -30,14 +31,8 @@ def call
 
       private
 
-      def validate_xms_mirid_format
-        begin
-          required_keys = %w(subscriptions resourcegroups providers)
-          missing_keys = required_keys - xms_mirid_hash.keys
-          raise Err::ClaimInInvalidFormat, "Required keys #{missing_keys} are missing" if missing_keys.empty?
-        rescue => e
-          raise Err::ClaimInInvalidFormat, e.inspect
-        end
+      def parse_xms_mirid
```

Comment:
why we need three function for it:
1. parse_xms_mirid call to 
2. xms_mirid_hash which call to
3. parsed_xms_mirid

i think that we can remove at least one function here

Reply:
Each one has its own purpose. The Command Class's `call` method needs to tell the story of what is going on - thus it has a call to the `parse_xms_mirid`.

`xms_mirid_hash` needs to be memoized so we don't do the work of `parsed_xms_mirid` every time. that's why it calls the `parsed_xms_mirid` method that does all the work and memoizes its output. 

the `parsed_xms_mirid` method does the actual parsing and has that name as in ruby we give methods that return an object the name of the returned object.

Reply:
Although the refactor is cleaner, was there a problem with the logic that this was refactored?

Reply:
Yes. we raised an error and rescued it straight after that so the original error was never raised